cmake/Modules/ValkeySetup.cmake

```diff
@@ -0,0 +1,349 @@
+    valkey_create_symlink(${target} ${link_name})
+endmacro ()
+
+# Return the current host distro name. For example: ubuntu, debian, amzn etc
+function (valkey_get_distro_name DISTRO_NAME)
+    if (LINUX AND NOT APPLE)
+        execute_process(
+            COMMAND /bin/bash "-c" "cat /etc/os-release |grep ^ID=|cut -d = -f 2"
+            OUTPUT_VARIABLE _OUT_VAR
+            OUTPUT_STRIP_TRAILING_WHITESPACE)
+        # clean the output
+        string(REPLACE "\"" "" _OUT_VAR "${_OUT_VAR}")
+        string(REPLACE "." "" _OUT_VAR "${_OUT_VAR}")
+        set(${DISTRO_NAME}
+            "${_OUT_VAR}"
+            PARENT_SCOPE)
+    elseif (APPLE)
+        set(${DISTRO_NAME}
+            "darwin"
+            PARENT_SCOPE)
+    elseif (IS_FREEBSD)
+        set(${DISTRO_NAME}
+            "freebsd"
+            PARENT_SCOPE)
+    else ()
+        set(${DISTRO_NAME}
+            "unknown"
+            PARENT_SCOPE)
+    endif ()
+endfunction ()
+
+# Determine if we are building in Release or Debug mode
+if (CMAKE_BUILD_TYPE MATCHES Debug OR CMAKE_BUILD_TYPE MATCHES DebugFull)
+    set(VALKEY_DEBUG_BUILD 1)
+    message(STATUS "Building in debug mode")
+else ()
+    set(VALKEY_DEBUG_BUILD 0)
+    message(STATUS "Building in release mode")
+endif ()
+
+# ----------------------------------------------------
+# Helper functions - end
+# ----------------------------------------------------
+
+# ----------------------------------------------------
+# Build options (allocator, tls, rdma et al)
+# ----------------------------------------------------
+
+if (NOT WITH_MALLOC)
+    if (APPLE)
+        set(WITH_MALLOC "libc")
+    elseif (UNIX)
+        set(WITH_MALLOC "jemalloc")
+    endif ()
+endif ()
+
+# User may pass different allocator library. Using -DWITH_MALLOC=<libname>, make sure it is a valid value
+if (WITH_MALLOC)
+    if ("${WITH_MALLOC}" STREQUAL "jemalloc")
+        set(MALLOC_LIB "jemalloc")
+        add_valkey_server_compiler_options("-DUSE_JEMALLOC")
+        set(USE_JEMALLOC 1)
+    elseif ("${WITH_MALLOC}" STREQUAL "libc")
+        set(MALLOC_LIB "libc")
+    elseif ("${WITH_MALLOC}" STREQUAL "tcmalloc")
+        set(MALLOC_LIB "tcmalloc")
+        add_valkey_server_compiler_options("-DUSE_TCMALLOC")
+    elseif ("${WITH_MALLOC}" STREQUAL "tcmalloc_minimal")
+        set(MALLOC_LIB "tcmalloc_minimal")
+        add_valkey_server_compiler_options("-DUSE_TCMALLOC")
+    else ()
+        message(FATAL_ERROR "WITH_MALLOC can be one of: jemalloc, libc, tcmalloc or tcmalloc_minimal")
+    endif ()
+endif ()
+
+message(STATUS "Using ${MALLOC_LIB}")
+
+# TLS support
+if (WITH_TLS)
+    find_package(OpenSSL REQUIRED)
+    message(STATUS "OpenSSL include dir: ${OPENSSL_INCLUDE_DIR}")
+    message(STATUS "OpenSSL libraries: ${OPENSSL_LIBRARIES}")
+
+    include_directories(${OPENSSL_INCLUDE_DIR})
+    add_valkey_server_compiler_options("-DUSE_OPENSSL=1")
+    add_valkey_server_compiler_options("-DBUILD_TLS_MODULE=0")
```

Comment:
so we will no longer build TLS as a module? It makes sense to me but I just wanted to make sure this is indeed intentional.